src/com/goide/intentions/GoMoveToStructInitializationIntention.java

```diff
@@ -0,0 +1,308 @@
+
+  @Contract("null, _-> false")
+  private static boolean isUninitializedFieldReferenceExpression(@Nullable GoReferenceExpression fieldReferenceExpression,
+                                                                 @NotNull GoCompositeLit structLiteral) {
+    if (fieldReferenceExpression == null) return false;
+    GoLiteralValue literalValue = structLiteral.getLiteralValue();
+    String fieldName = getIdentifierName(fieldReferenceExpression);
+    return literalValue != null &&
+           isFieldDefinition(fieldReferenceExpression.resolve()) &&
+           !ContainerUtil.exists(literalValue.getElementList(), element -> hasFieldName(element, fieldName));
+  }
+
+  @NotNull
+  private static Map<String, Integer> getIdentifierNameCounts(@NotNull List<GoReferenceExpression> fieldReferenceExpressions) {
+    return ContainerUtil.groupBy(fieldReferenceExpressions, GoMoveToStructInitializationIntention::getIdentifierName).entrySet().stream()
+      .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().size()));
+  }
+
+  @NotNull
+  private static String getIdentifierName(@NotNull GoReferenceExpression fieldReferenceExpression) {
+    return fieldReferenceExpression.getIdentifier().getText();
+  }
+
+  @NotNull
+  private static Collection<? extends PsiElement> getLeftHandElements(@NotNull GoStatement statement) {
+    if (statement instanceof GoSimpleStatement) {
+      GoShortVarDeclaration varDeclaration = ((GoSimpleStatement)statement).getShortVarDeclaration();
+      return varDeclaration != null ? varDeclaration.getVarDefinitionList() : ContainerUtil.emptyList();
+    }
+    if (statement instanceof GoAssignmentStatement) {
+      return ((GoAssignmentStatement)statement).getLeftHandExprList().getExpressionList();
+    }
+    return ContainerUtil.emptyList();
+  }
+
+  @Override
+  public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) throws IncorrectOperationException {
+    Pair<List<GoReferenceExpression>, GoCompositeLit> pair = getFieldReferencesAndStructLiteral(element);
+    if (pair.first == null || pair.first.isEmpty()) return;
+    moveFieldReferenceExpressions(pair.first, pair.second, project);
+  }
+
+  private static void moveFieldReferenceExpressions(@NotNull List<GoReferenceExpression> fieldReferenceExpressions,
+                                                    @NotNull GoCompositeLit structLiteral,
+                                                    @NotNull Project project) {
+    GoAssignmentStatement assignment = getValidAssignmentParent(ContainerUtil.getFirstItem(fieldReferenceExpressions));
+    if (assignment == null) return;
+
+    GoLiteralValue literalValue = structLiteral.getLiteralValue();
+    if (literalValue == null) return;
+
+    for (GoExpression expression : fieldReferenceExpressions) {
+      GoReferenceExpression fieldReferenceExpression = tryGetReferenceExpression(expression);
+      if (!isFieldReferenceExpression(fieldReferenceExpression)) continue;
```

Comment:
Why it can be not field reference expression? If it's matter add tests to cover it